packages/form-config/src/components/Antd/UploadForm/tests/index.test.tsx

```diff
@@ -0,0 +1,331 @@
+  onSaveRedirectURL: '/foo',
+  history,
+  location: {
+    hash: '',
+    pathname: `/upload-form`,
+    search: '',
+    state: {},
+  },
+  match: {
+    isExact: true,
+    params: { formId: null },
+    path: '/upload-form',
+    url: `/upload-form`,
+  },
+};
+
+describe('components/UploadFile', () => {
+  beforeAll(() => {
+    store.dispatch(
+      authenticateUser(
+        true,
+        {
+          email: 'bob@example.com',
+          name: 'Bobbie',
+          username: 'RobertBaratheon',
+        },
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        { api_token: 'hunter2', oAuth2Data: { access_token: 'hunter2', state: 'abcde' } }
+      )
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fetch.resetMocks();
+  });
+
+  it('renders without crashing', () => {
+    shallow(
+      <Provider store={store}>
+        <Router history={history}>
+          <UploadForm {...props} />
+        </Router>
+      </Provider>
+    );
+  });
+
+  it('renders correctly', () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <UploadForm {...props} />
+        </Router>
+      </Provider>
+    );
+
+    const content = wrapper.find('div.layout-content');
+    expect(content.find('Title').props()).toMatchSnapshot('title');
+    expect(content.find('Card').props()).toMatchSnapshot('card');
+    wrapper.unmount();
+  });
+
+  it('validates required fields', async () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <UploadForm {...props} />
+        </Router>
+      </Provider>
+    );
+
+    wrapper.find('form').simulate('submit');
+
+    await act(async () => {
+      await flushPromises();
+    });
+    wrapper.update();
+
+    expect(wrapper.find('FormItemInput').at(0).prop('errors')).toEqual(['Form Name is required']);
+    expect(wrapper.find('FormItemInput').at(1).prop('errors')).toEqual([]);
+    expect(wrapper.find('FormItemInput').at(2).prop('errors')).toEqual([]);
+    expect(wrapper.find('FormItemInput').at(3).prop('errors')).toEqual(['Form is required']);
+
+    wrapper.unmount();
+  });
+
+  it('uploads form', async () => {
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <UploadForm {...props} />
+        </Router>
+      </Provider>
+    );
+
+    wrapper
+      .find('input')
+      .at(0)
+      .simulate('change', { target: { name: 'form_name', value: 'test name' } });
+    wrapper
+      .find('input')
+      .at(1)
+      .simulate('change', { target: { name: 'module', value: 'test module' } });
+    wrapper
+      .find('input')
+      .at(2)
+      .simulate('change', { target: { name: 'form_relation', value: 'bar' } });
+    wrapper
+      .find('input')
+      .at(3)
+      .simulate('change', { target: { name: 'form', files: sampleFile } });
+    wrapper.find('form').simulate('submit');
+
```

Comment:
it would be easier for future modifications, if the inputs here were targeted by something like ids other than the index

Reply:
@p-netm I'll change that add use ids